index.rst

```diff
@@ -2,38 +2,192 @@
+  caches them on disk.
+
+When should you use ZODB?
+=========================
+
+When you want to focus on what your application does without database operations slowing you down
+  Even if find you need to incorporate or switch to another database
+  later, you can use ZODB in the early part of your project to make
+  initial discovery and learning much quicker.
+
+When your application has complex relationships and data structures
+  In relational databases you have to join tables to model complex
+  data structures and these joins can be tedious and expensive.  You
+  can mitigate this to some extent in databases like Postgres by using
+  more powerful data types like arrays and JSON columns, but when
+  relationships extend across rows, you still have to do joins.
+
+  In NoSQL databases, you can model complex data structures with
+  documents, but if you have relationships across documents, then you
+  have to do joins and join capabilities in NoSQL databases are
+  typically far less powerful and transactional semantics typically don't
+  cross documents, if they exist at all.
+
+  In ZODB, you can make objects as complex as you want and cross
+  object relationships are handled with Python object references.
+
+When the dominant access method is by accessing object attributes or by calling methods
+  If your primary object access is search, then other database
+  technologies might be a better fit.
+
+  ZODB has no query language other than Python. It's primary support
+  for search is through mapping object called BTrees.  People have
+  build higher-level search APIs on top of ZODB. These work well
+  enough to support some search.
+
+When you read data a lot more than you write it
+  ZODB caches aggressively, and if you're working set fits (or mostly
+  fits) in memory, performance is very good because it rarely has to
+  touch the database server.
+
+  If your application if very write heavy (e.g. logging), then you're
+  better off using something else.  Sometimes, you can use a database
+  suitable for heavy writes in combination with ZODB.
+
+When you want to write tests of application logic that interacts with your database
+  ZODB has a number of storage implementations, including layered
+  in-memory implementations that make testing very easy.
+
+  A database without an in-memory storage option can make testing very
+  complicated.
+
+When should you *not* use ZODB?
+===============================
+
+- When search is a dominant data access path
+
+- When you have high write volume
+
+- When caching is unlikely to benefit you
+
+  This can be the case when write volume is high, or when you tend to
+  access small amounts of data from a working set way too large to fit in
+  memory and when there's no good mechanism for dividing the working
+  set across application servers.
+
+- When you need non-Python tools to access your database, especially
+  tools designed to work with relational databases.
+
```

Comment:
I guess so. its another way of saying that unless your data has a good chance of already being in memory it may not be the right tool which you already said. (since it doesn't have an efficient way bring data into memory asynchronously ahead of when you need it)